packages/gboost-infra/src/file-upload/index.ts

```diff
@@ -0,0 +1,72 @@
+import { PolicyStatement } from "aws-cdk-lib/aws-iam";
+import { Construct } from "constructs";
+import { GraphqlApi } from "@aws-cdk/aws-appsync-alpha";
+import { Function } from "../function.js";
+import { createSchema } from "./createSchema.js";
+import { CommonProps, Stage } from "../common-props.js";
+import { Duration } from "aws-cdk-lib";
+
+interface FileUploadBucket {
+  bucket: string;
+  key: string;
+}
+
+export interface FileUploadProps extends CommonProps {
+  api: GraphqlApi;
+  /**
+   * AWS Partition where buckets exists
+   * @example "aws"
+   */
+  partition?: string;
+  /**
+   * Maps bucket keys to bucket names
+   * @example {
+   * BUCKET-KEY: { bucket: BUCKET-NAME, key: KEY },
+   * BUCKET-KEY2: { bucket: BUCKET-NAME2, key: KEY2}
+   * }
+   */
+  buckets: Record<string, FileUploadBucket>;
+}
+
+/**
+ * Creates a Lambda Function
+ */
+export class FileUpload extends Construct {
+  constructor(scope: Construct, id: string, props: FileUploadProps) {
+    super(scope, id);
+    const { api, stage = Stage.Dev, partition = "aws", buckets } = props;
+
+    const fileExt = import.meta.url.slice(-2);
+    const resources = [];
+    for (let i = 0; i < Object.keys(buckets).length; i++) {
+      const bucketKey = Object.keys(buckets)[i];
+      resources[2 * i] =
+        "arn:" + partition + ":s3:::" + buckets[bucketKey].bucket + "/*";
+      resources[2 * i + 1] =
+        "arn:" + partition + ":s3:::" + buckets[bucketKey].bucket;
+    }
```

Comment:
move below function instantiation and use `bucket.grantReadWrite(uploadFn)`

Reply:
Dynamically create nag suppression like here: https://github.com/cdklabs/cdk-nag/issues/957

Reply:
Using `grantReadWrite `and `grantDelete `works well. Still no nag errors. I'm confused as to where to put the nag suppression; My understanding was put it after I do `grantDelete `and `grantReadWrite`, but there is no `Stack` defined.

Reply:
Very weird. It must not be running.
You can use `Stack.of(this)` to access the stack.

Reply:
Oh, but the nag suppression should be put on the construct (bucket) not the stack level.

Reply:
Got it! Thanks!